### PR TITLE
feat(date): changed DateField and DateRangeField to allow using nodes for the label prop

### PR DESCRIPTION
**Pull request by stevenviar**

### CODE REVIEW

File: packages/date/src/DateField.js [hunk 1]
```diff
@@ -6,15 +6,19 @@ import Date from './Date';
 
 const DateField = ({ name, label, ...props }) => (
   <FormGroup for={name}>
-    {label && <Label for={name}>{label}</Label>}
+    {label && typeof label === 'string' ? (
```
Comment (GoPro16) [inline packages/date/src/DateField.js]: Looks like we missed some extra label attr spreading when adding in the new form library components.

The label type should look like the one linked below like reactstrap validation date:

https://github.com/Availity/availity-react/blob/master/packages/reactstrap-validation-date/src/AvDateRangeField.js#L129-L138

File: packages/date/src/DateRangeField.js [hunk 2]
```diff
@@ -6,15 +6,19 @@ import DateRange from './DateRange';
 
 const DateRangeField = ({ name, label, ...props }) => (
   <FormGroup for={name}>
-    {label && <Label for={name}>{label}</Label>}
+    {label && typeof label === 'string' ? (
```
Comment (GoPro16) [inline packages/date/src/DateRangeField.js]: Same thing for this field to have the proper label attrs and children.

**Comment by stevenviar** [comment]
Also, I noticed some of the tests for DateRange are failing today because it's trying to select the sibling to the current day on the calendar but it's the last day of the month.

**Comment by GoPro16** [comment]
> Also, I noticed some of the tests for DateRange are failing today because it's trying to select the sibling to the current day on the calendar but it's the last day of the month.

Interesting. Maybe we can modify it to get the previous sibling if the next doesn't exist or refactor how we are just picking the date as a whole.

**Pull request closed without merging**